Add missing includes for WTF modularization

adetaylor opened the pull request:
#### 7a0ace34a75c58cf3f3b7153835ec774ec4c9af5
<pre>
Add missing includes for WTF modularization
<a href="https://bugs.webkit.org/show_bug.cgi?id=292772">https://bugs.webkit.org/show_bug.cgi?id=292772</a>
<a href="https://rdar.apple.com/151008903">rdar://151008903</a>

Reviewed by Elliott Williams.

This adds various includes (in code and in a code generator) which are necessary to be able to interpret WebBackForwardList.h and some related files using Swift. Swift is stricter on C++ includes than existing WebKit C++ because it requires the C++ dependencies to be treated as clang modules; the module map for WTF (in a follow-up commit) will divide wtf into lots of small modules.

* Source/JavaScriptCore/assembler/SecureARM64EHashPins.h:
* Source/JavaScriptCore/bytecode/Watchpoint.h:
* Source/JavaScriptCore/heap/BlockDirectory.h:
* Source/JavaScriptCore/heap/Heap.h:
* Source/JavaScriptCore/heap/MarkedBlock.h:
* Source/JavaScriptCore/heap/WeakBlock.h:
* Source/JavaScriptCore/jit/ExecutableMemoryHandle.h:
* Source/JavaScriptCore/runtime/BufferMemoryHandle.h:
* Source/JavaScriptCore/runtime/ConcurrentJSLock.h:
* Source/JavaScriptCore/runtime/JSCJSValue.h:
* Source/JavaScriptCore/runtime/JSONAtomStringCache.h:
* Source/JavaScriptCore/runtime/JSString.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/WTF/wtf/CompletionHandler.h:
* Source/WTF/wtf/HashMap.h:
* Source/WTF/wtf/HashSet.h:
* Source/WTF/wtf/HashTable.h:
* Source/WTF/wtf/MallocSpan.h:
* Source/WTF/wtf/ThreadSafeWeakPtr.h:
* Source/WTF/wtf/WeakPtr.h:
* Source/WebCore/PAL/pal/PALSwift.h:
* Source/WebCore/loader/ResourceCryptographicDigest.h:
* Source/WebCore/page/csp/ContentSecurityPolicy.h:
* Source/WebCore/platform/ProcessQualified.h:
* Source/WebCore/platform/PublicSuffix.h:
* Source/WebCore/platform/RegistrableDomain.h:
* Source/WebCore/platform/SharedBuffer.h:
* Source/WebCore/platform/Timer.h:
* Source/WebCore/platform/graphics/FourCC.h:
* Source/WebCore/platform/network/HTTPHeaderMap.h:
* Source/WebCore/workers/service/ServiceWorkerTypes.h:
* Source/WebKit/Platform/IPC/Connection.h:
* Source/WebKit/Platform/IPC/Decoder.h:
* Source/WebKit/Platform/IPC/MessageReceiver.h:
* Source/WebKit/Platform/IPC/MessageReceiverMap.h:
* Source/WebKit/Platform/IPC/Timeout.h:
* Source/WebKit/Platform/IPC/cocoa/ImportanceAssertion.h:
* Source/WebKit/Platform/Logging.h:
* Source/WebKit/Platform/cocoa/WKMaterialHostingSupport.h:
* Source/WebKit/Scripts/webkit/messages.py:
(generate_message_names_header):
* Source/WebKit/Scripts/webkit/tests/MessageNames.h:
* Source/WebKit/Shared/API/APIObject.h:
* Source/WebKit/Shared/Cocoa/WKObject.h:
* Source/WebKit/Shared/MonotonicObjectIdentifier.h:
* Source/WebKit/Shared/WebPreferencesStore.h:
* Source/WebKit/UIProcess/API/APIFeature.h:
* Source/WebKit/UIProcess/API/cpp/WKRetainPtr.h:
* Source/WebKit/UIProcess/AuxiliaryProcessProxy.h:
* Source/WebKit/UIProcess/ProcessThrottler.h:
* Source/WebKit/UIProcess/ResponsivenessTimer.h:
* Source/WebKit/UIProcess/WebPreferences.h:

Canonical link: <a href="https://commits.webkit.org/294791@main">https://commits.webkit.org/294791@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/58e65ba60ae4310519c4007b31ccd8f802fd789e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102893 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/22567 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12887 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/108059 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/53534 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104932 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22887 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/31068 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78230 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35185 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105899 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17723 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92853 "Found 1 new API test failure: TestWebKitAPI.WebKit.PrintFrame (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58564 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/102369 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17563 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10889 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52891 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/95568 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/87372 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10956 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110434 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/101503 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/30030 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/22121 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87216 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/active-view-transition-type-on-non-root.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30394 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89049 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86835 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22136 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31681 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9396 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/24292 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29957 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/35279 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/125136 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29765 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34723 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/33092 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31327 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Platform/IPC/Decoder.h:
```diff
@@ -25,13 +25,16 @@
 #pragma once
 
+#include <bmalloc/BInline.h>
+#include <memory>
```
nit: these seem like they should be sorted down to L34.

emw-apple reviewed (approved):
LGTM. There is currently nothing in the build that will validate that these includes are present, right? Only future Swift-C++ code in WebKit?

adetaylor committed:
Add missing includes for WTF modularization
https://bugs.webkit.org/show_bug.cgi?id=292772
rdar://151008903

Reviewed by Elliott Williams.

This adds various includes (in code and in a code generator) which are necessary to be able to interpret WebBackForwardList.h and some related files using Swift. Swift is stricter on C++ includes than existing WebKit C++ because it requires the C++ dependencies to be treated as clang modules; the module map for WTF (in a follow-up commit) will divide wtf into lots of small modules.

* Source/JavaScriptCore/assembler/SecureARM64EHashPins.h:
* Source/JavaScriptCore/bytecode/Watchpoint.h:
* Source/JavaScriptCore/heap/BlockDirectory.h:
* Source/JavaScriptCore/heap/Heap.h:
* Source/JavaScriptCore/heap/MarkedBlock.h:
* Source/JavaScriptCore/heap/WeakBlock.h:
* Source/JavaScriptCore/jit/ExecutableMemoryHandle.h:
* Source/JavaScriptCore/runtime/BufferMemoryHandle.h:
* Source/JavaScriptCore/runtime/ConcurrentJSLock.h:
* Source/JavaScriptCore/runtime/JSCJSValue.h:
* Source/JavaScriptCore/runtime/JSONAtomStringCache.h:
* Source/JavaScriptCore/runtime/JSString.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/WTF/wtf/CompletionHandler.h:
* Source/WTF/wtf/HashMap.h:
* Source/WTF/wtf/HashSet.h:
* Source/WTF/wtf/HashTable.h:
* Source/WTF/wtf/MallocSpan.h:
* Source/WTF/wtf/ThreadSafeWeakPtr.h:
* Source/WTF/wtf/WeakPtr.h:
* Source/WebCore/PAL/pal/PALSwift.h:
* Source/WebCore/loader/ResourceCryptographicDigest.h:
* Source/WebCore/page/csp/ContentSecurityPolicy.h:
* Source/WebCore/platform/ProcessQualified.h:
* Source/WebCore/platform/PublicSuffix.h:
* Source/WebCore/platform/RegistrableDomain.h:
* Source/WebCore/platform/SharedBuffer.h:
* Source/WebCore/platform/Timer.h:
* Source/WebCore/platform/graphics/FourCC.h:
* Source/WebCore/platform/network/HTTPHeaderMap.h:
* Source/WebCore/workers/service/ServiceWorkerTypes.h:
* Source/WebKit/Platform/IPC/Connection.h:
* Source/WebKit/Platform/IPC/Decoder.h:
* Source/WebKit/Platform/IPC/MessageReceiver.h:
* Source/WebKit/Platform/IPC/MessageReceiverMap.h:
* Source/WebKit/Platform/IPC/Timeout.h:
* Source/WebKit/Platform/IPC/cocoa/ImportanceAssertion.h:
* Source/WebKit/Platform/Logging.h:
* Source/WebKit/Platform/cocoa/WKMaterialHostingSupport.h:
* Source/WebKit/Scripts/webkit/messages.py:
(generate_message_names_header):
* Source/WebKit/Scripts/webkit/tests/MessageNames.h:
* Source/WebKit/Shared/API/APIObject.h:
* Source/WebKit/Shared/Cocoa/WKObject.h:
* Source/WebKit/Shared/MonotonicObjectIdentifier.h:
* Source/WebKit/Shared/WebPreferencesStore.h:
* Source/WebKit/UIProcess/API/APIFeature.h:
* Source/WebKit/UIProcess/API/cpp/WKRetainPtr.h:
* Source/WebKit/UIProcess/AuxiliaryProcessProxy.h:
* Source/WebKit/UIProcess/ProcessThrottler.h:
* Source/WebKit/UIProcess/ResponsivenessTimer.h:
* Source/WebKit/UIProcess/WebPreferences.h:

Canonical link: https://commits.webkit.org/294791@main
